_posts/2019/2019-05-04-react-hooks-what-all-the-fuss-is-about.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: post
+title: React Hooks - What all the fuss is about
+date: 2019-05-04
+header_image: public/react-hooks.jpg
+header_position: center
+category: coding
+tags: ["javascript", "reactjs"]
+authors: ["Jonathan"]
+about_authors: ["jwieben"]
+---
+
+## React Hooks - What all the fuss is about
+
+Reacts new Hooks have been all the rage these last few months. You might have heard about them without really understanding what they are and why so many people are so excited about them. Maybe you have asked yourself why a simple API proposal would get [thousand of comments and reactions on its RFC](https://github.com/reactjs/rfcs/pull/68){:target="_blank"}. This post intends to help you understand some of the hype.
+
+#### What are Hooks?
+
+Hooks are an addition to React, that was released in February 2019. To explain what they do, I'll have to give some background on Reacts API.
+
+React lets developers split their UI into components. A component can do a variety of things: Render UI, keep a state or fetch data. **There are two ways of writing a component: As a function or as a class.** For the sake of simplicity, I will refer to those different types of components as function-components and class-components for the rest of this blog post.
+
+**Function-components** are pretty simple. They are just a JavaScript function that given a certain input (called "props" in React), return the UI that should be rendered on the screen.
+
+**Class-components**, on the other hand, are a bit more complex. They also take props and render UI, but on top of that keep a state and have different so-called lifecycle methods. So if, for instance, you wanted to update your state once some of your props changed, you would use a class-component to do that.
+
+And that's where hooks come in. **Hooks allow you to do all the things you previously needed classes for, inside a function-component.** 🎉
+
+#### What makes them so great?
+
+Now you might be asking yourself: What's the big deal? You can write your components in a different way, what's so crazy about that?
```

Comment:
what's the big deal (lowercase after colon)
You can write your components in a different way. What's so crazy about that?